### PR TITLE
Currency: Ensures correct number of decimal places for non-standard currencies

### LINKED ISSUE #1465: Currency: bitcoin decimal issue
According to https://duck.co/forum/thread/12013/btc-conversion-bug

I haven't tested it myself, but I wanted to call attention to this. 

------ 

IA Page: https://duck.co/ia/view/currency

### LINKED ISSUE #2724: Currency: Too few decimals for XBT
When calculating any currency into Bitcoins, there are too few decimals (at least for small amounts).
If I enter 0.20 EUR in XBT the answer is 0.00 XBT.

---

IA Page:  http://duck.co/ia/view/currency 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @MrChrisW

### PR DESCRIPTION
**Pull request by mkwia**
## Description of new Instant Answer, or changes

Fixes instances of currencies with non-standard number of decimals geting rounded to 2 decimal places.

## Related Issues and Discussions

Fixes #1465 and #2724 

## People to notify

@javathunderman @MonoMonkey @MrChrisW

---

Instant Answer Page: https://duck.co/ia/view/currency

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/currency/currency.js [hunk 1]
```diff
@@ -95,11 +100,17 @@
...
         // Add commas to the numbers for display.
-        function formatNumber(x) {
+        function formatNumber(x, symbol) {
             // Check if the number has a decimal point.
-            // If it does, only show the first two digits after the decimal place.
+            // If it does, only show the standard number digits after the decimal place for a given currency.
+            symbol = symbol.toLowerCase();
             if(/\./.test(x.toString())) {
-                x = x.toFixed(2);   
+                if (symbol in currency_nonstandard_decimal) {
```
Comment (moollaza) [inline share/spice/currency/currency.js]: nit: I would just set a precision variable with a default value. Then call `toFixed()` once:

```js
var precision = currency_nonstandard_decimal[symbol] || 2;
x = x.toFixed(precision);
```

File: share/spice/currency/currency.js [hunk 2]
```diff
@@ -24,6 +24,11 @@
         'gb': 'uk'
     };
     
+    // Currencies with non-standard decimal points. (Standard is 2 eg. 10.46)
+    var currency_nonstandard_decimal = {
```
Comment (moollaza) [inline share/spice/currency/currency.js]: let's shorten the name here. I think `decimal_places` is fine.

**Comment by moollaza** [comment]
Deploying to Beta 👍

**Comment by moollaza** [comment]
@mkwia Thanks! This LGTM :+1: Great job 🚀

**Pull request merged**